webrender/src/resource_cache.rs

```diff
@@ -149,24 +149,23 @@ pub struct ResourceClassCache<K: Hash + Eq, V, U: Default> {
 
 pub fn intersect_for_tile(
     dirty: DeviceUintRect,
-    width: u32,
-    height: u32,
+    actual_tile_size: DeviceUintSize,
     tile_size: TileSize,
     tile_offset: TileOffset,
 
 ) -> Option<DeviceUintRect> {
-        dirty.intersection(&DeviceUintRect::new(
-            DeviceUintPoint::new(
-                tile_offset.x as u32 * tile_size as u32,
-                tile_offset.y as u32 * tile_size as u32
-            ),
-            DeviceUintSize::new(width, height),
-        )).map(|mut r| {
-                // we can't translate by a negative size so do it manually
-                r.origin.x -= tile_offset.x as u32 * tile_size as u32;
-                r.origin.y -= tile_offset.y as u32 * tile_size as u32;
-                r
-            })
+    dirty.intersection(&DeviceUintRect::new(
+        DeviceUintPoint::new(
+            tile_offset.x as u32 * tile_size as u32,
```

Comment:
any plans to vectorize this? :)

Reply:
I am not not sure what you mean here. `tile_offset` is a u16 "in number of tiles" vector, which we multiply by a scalar. Are your referring to something like creating a `ScaleFactor<u32, Tiles, DeviceSpace>` from `tile_size as u32` and multiply `tile_offset`?

Reply:
Don't worry about this, but I was thinking about something like:
```rust
dirty.intersection(&DeviceUintRect::new(tile_offset.to_u32() * tile_size as u32, actual_tile_size)
    .map(|r| r.translate_neg(tile_offset.to_u32() * tile_size as u32))
            
```

Reply:
It'd also need tile_offset to be turned into a vector in device space instead of a point in tiles for the thing to chain properly.